app/src/main/java/com/example/itbook/ui/home/HomePresenter.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.itbook.ui.home
+
+import com.example.itbook.data.model.Book
+import com.example.itbook.data.model.PreviewCategory
+import com.example.itbook.data.repository.BooksRepository
+import com.example.itbook.data.source.remote.API.APIQuery
+import com.example.itbook.utils.Categories
+import com.example.itbook.utils.OnDataLoadCallBack
+
+const val SIZE_PREVIEW_CATEGORIES = 5
+
+class HomePresenter(
+    private val view: HomeContract.View,
+    private val repository: BooksRepository
+) : HomeContract.Presenter {
+
+    private val categories = Categories.getCategories()
+    private val previewCategories = mutableListOf<PreviewCategory>()
+    private var error = ""
+
+    override fun start() {
+
+        getCategories()
+        getPreviewCategories()
+    }
+
+    override fun getCategories() {
+        view.showCategories(categories)
+    }
+
+    override fun getPreviewCategories() {
+        getNewBooks()
+        getBooksByCategories()
+    }
+
+    fun getNewBooks() {
+        view.showLoading(true)
+        repository.getRemoteBooks(APIQuery.queryNews(), object : OnDataLoadCallBack<List<Book>> {
+            override fun onSuccess(data: List<Book>) {
+                previewCategories.add(
+                    PreviewCategory(
+                        categories[0],
+                        data as MutableList<Book>
```

Comment:
```suggestion
                        data.toMutableList()
```
dùng hàm có sẵn chứ không nên cast